runtime/src/snapshot_utils.rs

```diff
@@ -643,7 +643,7 @@ pub fn archive_snapshot_package(
         )
     })?;
 
-    let src_snapshot_dir = snapshot_package.snapshot_links.join(&slot_str);
+    let src_snapshot_dir = &snapshot_package.bank_snapshot_dir;
```

Comment:
I changed this field from bank_snapshots_dir (with s) to bank_snapshot_dir (without s).  In the calling part, the parent() call is also removed.  There is no more need to pass the top snapshots dir, because all the info about this specific snapshot at this slot is under the specific snapshot dir.

Reply:
> Looks like there are a few places that change the value of what `snapshot_links` (or the new field) path is. Please do those changes in a separate PR from a simple renaming.

There are two parts of changes. 
1. rename snapshot_links to bank_snapshots_dir (with s);
2. change to the slot-specific bank_snapshot_dir (withtout s);

The 2 is to remove some unnecessary operation (the caller part calling parent(), and in the function derive bank_snapshot_dir from the top snapshot dir).

Reply:
Earlier the very beginning it was the top directory of the TempDir created for the AccountsPackage.

From the master, I checked out this version 6f9a15c36b 
```
* 16f3dcd5d2 Update fn create_and_verify_snapshot (#31245)
* 6f9a15c36b (HEAD) ci: suggest to use hidden_unless_forced() (#31321)
```

I ran the test test_package_snapshots_relative_ledger_path, break at the line 290 when SnapshotPackage is composed with its snapshot_links, and dumped the files under it.
```
xiangzhu@Xiang-Work-Mac split_pr % tree "/Users/xiangzhu/solana/solana/farf/tmp/test_package_snapshots/.tmpzW8LaQ"
/Users/xiangzhu/solana/solana/farf/tmp/test_package_snapshots/.tmpzW8LaQ
├── 0
│   └── 0
├── 1
│   └── 1
├── 2
│   └── 2
├── 3
│   └── 3
└── 4
    └── 4
5 directories, 5 files
```
It has the same layout as the snapshots top directory (bank_snapshots_dir with s).

In the PR https://github.com/solana-labs/solana/pull/31245/files, the left side is the old version of fn create_and_verify_snapshot().  The for loop at line 272 is where such layout is generated.  link_snapshots_dir at line 271 is the outer-scope variable when goes into the SnapshotPackage.  Inside the for loop, there is a inner-scope link_snapshots_dir which is the sub directory with slot (same as bank_snapshot_dir without s).